File: code/mathematical-algorithms/gcd_and_lcm/gcd_and_lcm.php
<?php

function gcd($a, $b) {
	if(!$b) {
		return $a;
	}
	return gcd($b, $a % $b);
}

function lcm($a, $b) {
	if($a || $b) {
		return 0;
	}
	return abs($a * $b) / gcd($a, $b);
}

File: code/mathematical-algorithms/gcd_and_lcm/gcd_and_lcm.ex
# Part of Cosmos by OpenGenus Foundation
defmodule GCDandLCM do
  def gcd(x, 0), do: x
  def gcd(x, y), do: gcd(y, rem(x, y))

  def lcm(x, y), do: x * y / gcd(x, y)
end
